File: apps/student/src/main/java/com/instructure/student/util/AnalyticsEventHandling.java
package com.instructure.student.util;

public interface AnalyticsEventHandling {
    void trackButtonPressed(String buttonName, Long buttonValue);
    void trackScreen(String screenName);
    void trackEnrollment(String enrollmentType);
    void trackDomain(String domain);
    void trackEvent(String category, String action, String label, long value);
    void trackUIEvent(String action, String label, long value);
    void trackTiming(String category, String name, String label, long duration);
}

File: apps/student/src/main/java/com/emeritus/student/util/AnalyticsEventHandling.kt
package com.emeritus.student.util

interface AnalyticsEventHandling {
    fun trackButtonPressed(buttonName: String?, buttonValue: Long?)
    fun trackScreen(screenName: String?)
    fun trackEnrollment(enrollmentType: String?)
    fun trackDomain(domain: String?)
    fun trackEvent(category: String?, action: String?, label: String?, value: Long)
    fun trackUIEvent(action: String?, label: String?, value: Long)
    fun trackTiming(category: String?, name: String?, label: String?, duration: Long)
}
